### PR TITLE
Remove PSI from QuickFix for undefined callback functions

### PR DESCRIPTION
**Pull request by telezhnaya**

### CODE REVIEW

File: src/org/intellij/erlang/inspection/ErlangUndefinedCallbackFunctionInspection.java [hunk 1]
```diff
@@ -64,26 +65,30 @@ protected void checkFile(@NotNull ErlangFile file, @NotNull ProblemsHolder probl
...
       }
       String message = builder.substring(0, builder.length() - 2);
-      registerProblem(problemsHolder, behaviourRef, message, new MyLocalQuickFixBase(undefinedCallbacks));
+      registerProblem(problemsHolder, behaviourRef, message, new CreateAndExportFunctionsFix(undefinedCallbacks));
     }
   }
 
-  private static class MyLocalQuickFixBase extends LocalQuickFixBase {
+  private static class CreateAndExportFunctionsFix extends LocalQuickFixBase {
     @NotNull
```
Comment (deadok22) [inline src/org/intellij/erlang/inspection/ErlangUndefinedCallbackFunctionInspection.java]: Do not annotate fields unless they're mutable.

File: src/org/intellij/erlang/inspection/ErlangUndefinedCallbackFunctionInspection.java [hunk 2]
```diff
@@ -64,26 +65,30 @@ protected void checkFile(@NotNull ErlangFile file, @NotNull ProblemsHolder probl
...
-    private final Collection<ErlangCallbackSpec> myCallbackSpecs;
+    private final Collection<SmartPsiElementPointer<ErlangCallbackSpec>> myCallbackSpecs;
```
Comment (deadok22) [inline src/org/intellij/erlang/inspection/ErlangUndefinedCallbackFunctionInspection.java]: Please, avoid using Collection. You iterate over it, expecting some concrete iteration order. Collection doesn't bring you this property. Why not use list instead?

File: src/org/intellij/erlang/inspection/ErlangUndefinedCallbackFunctionInspection.java [hunk 3]
```diff
@@ -64,26 +65,30 @@ protected void checkFile(@NotNull ErlangFile file, @NotNull ProblemsHolder probl
...
 
-    protected MyLocalQuickFixBase(@NotNull Collection<ErlangCallbackSpec> callbackSpecs) {
+    public CreateAndExportFunctionsFix(@NotNull Collection<ErlangCallbackSpec> callbackSpecs) {
       super(FIX_MESSAGE);
-      myCallbackSpecs = callbackSpecs;
+      myCallbackSpecs = ContainerUtil.map(callbackSpecs, new Function<ErlangCallbackSpec, SmartPsiElementPointer<ErlangCallbackSpec>>() {
+        @Override
+        public SmartPsiElementPointer<ErlangCallbackSpec> fun(ErlangCallbackSpec callbackSpec) {
+          return SmartPointerManager.getInstance(callbackSpec.getProject()).createSmartPsiElementPointer(callbackSpec);
```
Comment (deadok22) [inline src/org/intellij/erlang/inspection/ErlangUndefinedCallbackFunctionInspection.java]: You only need to obtain SmartPointerManager instance once. Why doing it for each callback you are to implement?

**Comment by telezhnaya** [comment]
Check it please.

**Pull request merged**